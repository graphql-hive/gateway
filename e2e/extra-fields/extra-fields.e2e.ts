import { createTenv } from '@internal/e2e';
import { expect, it } from 'vitest';

const { gateway, compose, service } = createTenv(__dirname);

it('resolves extra fields', async () => {
  const { output } = await compose({
    services: [await service('foo'), await service('bar')],
    output: 'graphql',
  });
  const { execute } = await gateway({ supergraph: output });
  await expect(
    execute({
      query: /* GraphQL */ `
        query FooBarFoo {
          foo {
            id
            bar {
              id
              foo {
                id
              }
            }
          }
        }
      `,
    }),
  ).resolves.toMatchSnapshot();
});
