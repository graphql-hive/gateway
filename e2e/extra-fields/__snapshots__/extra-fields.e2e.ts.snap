// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`resolves extra fields 1`] = `
{
  "data": {
    "foo": {
      "bar": {
        "foo": {
          "id": "1",
        },
        "id": "1",
      },
      "id": "1",
    },
  },
}
`;
